hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -35,11 +35,20 @@
       EnrichedSpanConstants.getValue(BoundaryTypeValue.BOUNDARY_TYPE_VALUE_ENTRY);
   private static final String EXIT_BOUNDARY_TYPE =
       EnrichedSpanConstants.getValue(BoundaryTypeValue.BOUNDARY_TYPE_VALUE_EXIT);
+  private static final String X_FORWARDED_HOST_SERVER = "x-forwarded-server";
+  private static final String X_FORWARDED_HOST_HEADER = "x-forwarded-host";
+
   private static final List<String> HOST_HEADER_ATTRIBUTES = ImmutableList.of(
+      // The order of these constants is important because that enforces the priority for
+      // different keys/headers.
       RawSpanConstants.getValue(org.hypertrace.core.span.constants.v1.Http.HTTP_REQUEST_HOST_HEADER),
       RawSpanConstants.getValue(org.hypertrace.core.span.constants.v1.Http.HTTP_REQUEST_AUTHORITY_HEADER),
-      RawSpanConstants.getValue(org.hypertrace.core.span.constants.v1.Http.HTTP_HOST)
+      RawSpanConstants.getValue(org.hypertrace.core.span.constants.v1.Http.HTTP_HOST),
+      // In the cases where there are sidecar proxies, the host header might be set to localhost
+      // while the original host will be moved to x-forwarded headers. Hence, read them too.
+      X_FORWARDED_HOST_SERVER, X_FORWARDED_HOST_HEADER
```

Comment:
I think we should prefer `X_FORWARDED_HOST_HEADER` over `X_FORWARDED_HOST_SERVER`. `X_FORWARDED_HOST_HEADER` is the true value of the host, whereas proxy can choose to overwrite `X_FORWARDED_HOST_SERVER`

Reply:
Nevermind. Saw your comment